### ISSUE TITLE
[Feature Request] aria-label="Scroll to top button"

### ISSUE THREAD

**Issue by massimoghm** [bootscore/bootscore#916]
### Describe the feature that you'd like

Hi,

for better accessibility, I think it's a good idea to add an aria-label="Go to top" with a translatable string to the "back to top" button.

`<a href="#" class="<?= apply_filters('bootscore/class/footer/to_top_button', 'btn btn-primary shadow'); ?> position-fixed zi-1000 top-button" aria-label="<?php esc_html_e( 'Go to top', 'bootscore' ); ?>"><i class="fa-solid fa-chevron-up"></i><span class="visually-hidden-focusable">To top</span></a>`

I think is also a good choice add to header this two shortcut.

```
<a class="skip-link visually-hidden-focusable" href="#primary"><?php esc_html_e( 'Go to the content', 'bootscore' ); ?></a>
<a class="skip-link visually-hidden-focusable" href="#footer"><?php esc_html_e( 'Go to the footer', 'bootscore' ); ?></a>
```
With this or similar CSS:

`a.skip-link.visually-hidden-focusable {
    top: 15px;
    left: 15px;
    padding: 15px;
    color: #000;
    z-index: 999;
    font-weight: 600;
    position: absolute;
    width: fit-content;
    background: #dfdfdf;
}`

Best regards!

### Why is this feature needed? What problem does it solve?

Accessibility fix.

### Checklist

- [X] I have searched the issues and this feature has not been requested.

**Comment by massimoghm** [comment]
Hi @crftwrk,

Thank you for the points raised.
We could use this approach to allow for customization in the child.

`<div class="skippy visually-hidden-focusable">
    <a class="visually-hidden-focusable" href="#primary"><?php esc_html_e( 'Skip to main content', 'bootscore' ); ?></a>
    <a class="visually-hidden-focusable" href="#footer"><?php esc_html_e( 'Skip to the footer', 'bootscore' ); ?></a>
  </div>`
  
It wouldn't display the buttons inline as, for example, happens on the Bootstrap documentation site, but it would still be possible to style each link individually.

For the footer ID, can we use "footer" or could it cause issues?

What do you think?

Regards!

**Comment by crftwrk** [comment]
Think an outer `div` is not needed, we can add a `skip-link` class (sounds a bit better than `skippy` in my opinion ;-)

```php
 <a class="visually-hidden-focusable skip-link" href="#primary"><?php esc_html_e( 'Skip to main content', 'bootscore' ); ?></a>
 <a class="visually-hidden-focusable skip-link" href="#bootscore-footer"><?php esc_html_e( 'Skip to the footer', 'bootscore' ); ?></a>
```

To link to the footer, this needs an unique ID. Simply add `id="bootscore-footer"` to this line:

https://github.com/bootscore/bootscore/blob/901372fe9d1390531f468941eafeaf991f06b53a/footer.php#L20

`<footer id="bootscore-footer" class="bootscore-footer"> `

So, in this case the id and class have the same name, but we are not able to change class without crashing existing sites.

What do you think?

**Comment by massimoghm** [comment]
> Hi,
> 
> thank you for pointing this out, this is a good idea! Some thoughts:
> 
> * `footer` has no ID yet, we have to add one
> * Using the translatable string for `visually-hidden` `span` as well `<a href="#" class="<?= apply_filters('bootscore/class/footer/to_top_button', 'btn btn-primary shadow'); ?> position-fixed zi-1000 top-button" aria-label="<?php esc_html_e( 'Go to top', 'bootscore' ); ?>"><i class="fa-solid fa-chevron-up"></i><span class="visually-hidden-focusable"><?php esc_html_e( 'Go to top', 'bootscore' ); ?></span></a>`
> * No additional CSS, because Bootscore use Bootstrap as it is, see https://getbootstrap.com/docs/5.3/helpers/visually-hidden/ what this class exactly does. Individual styles should be added to the child-theme.
> 
> Do you want to contribute and create a PR? If you need help here, I will walk you through the process.

Can you walk me through the process of pull request?

Regards!

**Comment by crftwrk** [comment]
Yes, that is simple:

- Go to https://github.com/bootscore/bootscore and press the Fork button on the top right.
- Now you own a copy of the theme.
- Do changes in header.php and footer.php. GitHub will create automatically a new branch.
- When changes done, GitHub will ask you if you want to create a Pull request. Press Yes and merge branch into your main branch.
- GitHub will ask you again if you want to create a pull request merging your main branch into Bootscore's main. Press Yes, that's it.

I'm here if you need more help.
